### DISCUSSION TITLE
Serverless SMTP SEND EMAIL Function

### DISCUSSION THREAD

**Discussion by landed1** [vercel/vercel#6250] [General]
Been searching for a few days now and don't see anything that answers my questions through a web search other engines as well as the duckduckgo used.
I am trying to build a static web form and just want to email the content to an email address.

From the docs I am encouraged to use an SMTP API such as mailchimp. But no sample code on this.

I see that I can write javascript in the api folder under a class. Is this what it means to be serverless. I am not sure why I need the api route in the first place. Can't I just import node mailer in the app and send direct from there?

It's a silly question for experienced devs I get that. I also was worried that I need to hide the API key secret. 

Thanks for heads up or where to go to learn what I am missing. I have an SMTP account I can use to send the mail from. Is it just not possible to call that from the front end. Is this why I need the api route as this is then on the server?

I also read a thread that said SMTP was being blocked upstream so just wanted to get some clarity thanks.

**Comment by landed1** [comment]
This is the closest I have got so far from Kenny- https://kennymark.com/blog/how-to-send-serverless-emails-with-next-js-and-sengrid-40lj
Seems nodemailer is not great on production so sendgrid is a way out that seems to work, untested by myself at this point.

Form submits to a route (route1) via POST
route 1 Capture the form data into an object and call route 2 (not sure why this isn't just a function in the app)
route 2 uses the sendgrid library and their API to actually send the email.

**Comment by barelyhuman** [comment]
if you don't really have any preferences, you can hit on https://mailer.reaper.im/ as an API call and it'll take care of sending through the email , if you plan on receiving email, the same applies, though you might wanna mask the mailer url with a function in next.js or go function or whatever you are using, to avoid leaking out the password and everything while making a request. 

I normally leave the url in the environment variables. 
Hope that helps.

**Comment by leerob** [comment]
You can find more guidance on sending emails from Serverless Functions here 😄 https://vercel.com/docs/solutions/email
